plugins/Marketplace/PackageList.py

```diff
@@ -23,164 +14,80 @@
 
 
 class PackageList(ListModel):
+    """ A List model for Packages, this class serves as parent class for more detailed implementations.
+    such as Packages obtained from Remote or Local source
     """
-    Represents a list of packages to be displayed in the interface.
-
-    The list can be filtered (e.g. on package type, materials vs. plug-ins) and
-    paginated.
-    """
-
     PackageRole = Qt.UserRole + 1
 
-    ITEMS_PER_PAGE = 20  # Pagination of number of elements to download at once.
-
     def __init__(self, parent: "QObject" = None) -> None:
         super().__init__(parent)
-
-        self._ongoing_request: Optional[HttpRequestData] = None
-        self._scope = JsonDecoratorScope(UltimakerCloudScope(CuraApplication.getInstance()))
         self._error_message = ""
-
-        self._package_type_filter = ""
-        self._request_url = self._initialRequestUrl()
-
         self.addRoleName(self.PackageRole, "package")
-
-    def __del__(self) -> None:
-        """
-        When deleting this object, abort the request so that we don't get a callback from it later on a deleted C++
-        object.
-        """
-        self.abortRequest()
+        self._is_loading = False
+        self._has_more = False
+        self._has_footer = True
 
     @pyqtSlot()
-    def request(self) -> None:
-        """
-        Make a request for the first paginated page of packages.
-
-        When the request is done, the list will get updated with the new package models.
-        """
-        self.setErrorMessage("")  # Clear any previous errors.
-
-        http = HttpRequestManager.getInstance()
-        self._ongoing_request = http.get(
-            self._request_url,
-            scope = self._scope,
-            callback = self._parseResponse,
-            error_callback = self._onError
-        )
-        self.isLoadingChanged.emit()
+    def updatePackages(self) -> None:
+        """ A Qt slot which will update the List from a source. Actual implementation should be done in the child class"""
+        pass
 
     @pyqtSlot()
-    def abortRequest(self) -> None:
-        HttpRequestManager.getInstance().abortRequest(self._ongoing_request)
-        self._ongoing_request = None
+    def abortUpdating(self) -> None:
+        """ A Qt slot which allows the update process to be aborted. Override this for child classes with async/callback
+        updatePackges methods"""
+        pass
 
     def reset(self) -> None:
+        """ Resets and clears the list"""
         self.clear()
-        self._request_url = self._initialRequestUrl()
 
-    isLoadingChanged = pyqtSignal()
+    isLoadingChanged = pyqtSignal()  # The signal for isLoading property
```

Comment:
```suggestion
    isLoadingChanged = pyqtSignal() 
```

Up for debate; But we use {property}Changed as the signal for the change of the property all over the code, so it's perhaps a bit much to re-document this.

Reply:
There are a few other cases as well, but I didn't want to put a remark for all of them. Those should still be resolved.